### ISSUE TITLE
Assign Issues / Pull Requests to Teams

### ISSUE THREAD

**Issue by aensley** [dear-github/dear-github#170]
Related to #131.

It's really nice that we can now assign issues and pull requests to multiple people. What would really be killer would be the ability to assign issues and pull requests to teams (e.g. developers, analysts, reviewers, testers, system administrators, deployment managers, etc.).

The concept I envision would be like the **Open** and **Closed** tabs on the **Milestone** dropdown. Just add two tabs to the **Assignees** dropdown: **People** | **Teams**

**Comment by pandora** [comment]
+1 
Review Board allows users to assign reviews to both individuals and groups. Having the ability to do this on Github would make it possible to assign PR's to certain primary individuals as well as a broader group of people.

**Comment by brianmrock** [comment]
+1 this would be a great feature to add!

**Comment by JKomoroski** [comment]
bump

**Comment by jakelheknight** [comment]
+15 I have a big team and we all want this feature. :)

**Comment by leemgs** [comment]
+50 Our team also wants to use a great feature.

**Comment by SimenB** [comment]
It is available now.

![image](https://user-images.githubusercontent.com/1404810/31986315-50f3b866-b968-11e7-9649-a843a78c3ac4.png)

**Comment by leemgs** [comment]
@SimenB Does the Github Enterprise  version supports the fuction that you commented?

**Comment by SimenB** [comment]
Did you try to check this yourself?

It's available in GH enterprise 2.11, released in september

https://github.com/blog/2429-introducing-github-enterprise-2-11

**Comment by leemgs** [comment]
Aha  We have beeing used version 2.10 of Github Ent. Thanks

**Comment by aensley** [comment]
_Replying to https://github.com/dear-github/dear-github/issues/170#issuecomment-339242536 by @SimenB:_

That's Reviewers; not Assignees. Ironically, asking for a whole team to review a PR is useless to us, while being able to assign a PR to the "Testers" team is basically why I wrote this issue.

**Comment by tsal** [comment]
I have to agree. Assignees would be much more useful.

**Comment by merland** [comment]
It would be really nice to be able to assign a whole team to issues and PRs. It comes down to psychology and group dynamics. 

In my view, it's very hard to achieve self-organization in a team if issues are assigned to individuals. 
The whole team is responsible for every issue, not a single person. This guiding principle should of course be reflected in the tooling :) 

May be less applicable to distributed teams, idk.

**Comment by burtonator** [comment]
Good god.. this issue is super important.

**Comment by aghassemi** [comment]
Please!

**Comment by PVince81** [comment]
Slightly related, would also be useful: https://github.com/dear-github/dear-github/issues/306

**Comment by brandonseroyer** [comment]
+1

This feature would be very useful

**Comment by themitchy** [comment]
This works... kind of. You can assign a team as a reviewer and the appropriate notifications go out. However, as soon as a review is posted, the team is removed from the "requested reviewers" field and replaced with the user that submitted a review.

**Comment by groke** [comment]
+1

**Comment by JDecker18** [comment]
40+ on our repo. Need this for organization.
![image](https://user-images.githubusercontent.com/40579216/50022735-cc2baf80-ffa2-11e8-89a8-90c5f895a668.png)

**Comment by Lownin** [comment]
This would be great

**Comment by CraigLager** [comment]
+1

**Comment by borick** [comment]
+1

the behavior of re-assigning to particular user is kind of annoying

**Comment by undrline** [comment]
It's been three years.  Is there another way to get traction on this?

**Comment by DenisBBC** [comment]
Up

**Comment by DenisBBC** [comment]
Any updates on this?

**Comment by edouardberthe** [comment]
This would indeed be a very useful feature.
Could we have any update?

**Comment by patcon** [comment]
A probot plugin could probably be used to do this in the meantime

**Comment by Salstar24** [comment]
+1 for this idea, would really help in my company.

**Comment by TheDers** [comment]
+1!

**Comment by HolySamosa** [comment]
+100.  This would simplify our workflow immensely!

**Comment by tehsphinx** [comment]
+1

**Comment by SoManyHs** [comment]
Agreed. Would be especially helpful if there could be an email associated with a team, which might be shared across a team (e.g. oncall-engineer@mycompany.com).

**Comment by ankurja** [comment]
Is there any update on this ? I am using GitHub Enterprise 2.18.5 and this option is still not available.

**Comment by ADWolfe-LoyNO** [comment]
> It is available now, and has been for some time.
> 
> ![image](https://user-images.githubusercontent.com/1404810/31986315-50f3b866-b968-11e7-9649-a843a78c3ac4.png)

Reviews are ok, but issues are how I want to track bugs and I can't assign issues to teams. Please???

**Comment by undrline** [comment]
> It is available now, and has been for some time.
> Reviews are ok, but issues are how I want to track bugs and I can't assign issues to teams. 

That second line is the point ...
https://github.com/dear-github/dear-github/issues/170#issuecomment-339805825
https://github.com/dear-github/dear-github/issues/170#issuecomment-445405700
https://github.com/dear-github/dear-github/issues/170#issuecomment-447427052

**Comment by halfwit** [comment]
Seems like the one killer feature that is missing from all of this team/issue setup.

**Comment by dennisholmer** [comment]
For pull requests it's possible to assign team to "Reviewers", but not to "Assignees". Would be nice to be able to assign teams to assignees too, currently we have to select multiple people as assignees all the time.

**Comment by halfwit** [comment]
I do get that the team is supposed to assign from inside itself, but for a project manager to assign to a team initially would be much better. We can always `@myproject/team` but it's clumsy.

**Comment by ardrigh** [comment]
How is this still not available feature? 

Please stop adding all these other `beta` features and add the basic workflow like this one!

**Comment by arash009** [comment]
plus a million. We're being forced to use labels to solve this.

**Comment by guilyx** [comment]
This would actually be really useful, +alot

**Comment by jercle** [comment]
+1 would be very useful

**Comment by ghost** [comment]
+1, I had assumed since it worked on reviews it would work on issue assignments, but I was wrong.
This would be very useful. Our use case is that we want to let users assign feature requests to `product` team and all PMs get notified so they can choose to assign it to someone.

**Comment by pmyjavec** [comment]
This is very confusing that it's supported for PRs but not Issues, took me a while to workout what's happening.

**Comment by amadib** [comment]
Are there plans to support this in the future?

**Comment by ardrigh** [comment]
It's 2020 and this request is still open :ghost:

**Comment by rgoitia** [comment]
I cannot believe this is not a feature and there is no official answer to this request. This is just mind blowing 😮 .

**Comment by JFQueralt** [comment]
Octocat, could you purr on the team's lap on this one to see if they can work on it?

**Comment by joelamajors** [comment]
I also would like to have this feature.

**Comment by mrthrowawaylabs** [comment]
I'd like this feature as well.

**Comment by julio8a** [comment]
Yes, please!

**Comment by joshua-reddish** [comment]
Truly incredible we dont have this yet tbh. This ticket is almost 6 years old

**Comment by richardtru** [comment]
+1 would love to have this available!

**Comment by whanata** [comment]
+1 on having this also.

**Comment by pradeepvenkatakrishnan** [comment]
+1 would love to have this available!

**Comment by mlekoo** [comment]
+1

**Comment by stuart909** [comment]
+1

**Comment by kaanchan** [comment]
+1 Essential for non-profit teams

**Comment by levinebw** [comment]
++ Allow to assign issues to a team

**Comment by aidanheerdegen** [comment]
+1 for our team (not that these seem to be making much difference).

Not only would it be super useful, but it makes the UI consistent. Such a waste of time to try this, and different combinations, check you've type the team name correctly, give up. search, find this issue and discover this is not consistent with the functionality of reviewers.

**Comment by dmkim22-lguplus** [comment]
+1
so even after more than 5 years with consistent requests.. there's no plan on this?

**Comment by APIStacey** [comment]
Yes.. really need this. Why can I not do this, but I can tag in a comment on an issue.

**Comment by big-andy-coates** [comment]
I may as well add a +1 as well...

**Comment by brantjwalker** [comment]
+1

**Comment by angeleg** [comment]
+1. Super counter-intuitive that this does not work the way you expect it to.

**Comment by zafersevik-productmadness** [comment]
+1

**Comment by rpsene** [comment]
+1

**Comment by polinabee** [comment]
Any update on this? I am surprised that you cannot assign an issue to a team, especially since you can assign a PR to a team. I need this feature for proper triage of issues by contributors

**Comment by badgeminer** [comment]
+1

**Comment by momomomo111** [comment]
+1

**Comment by ViacheslavKudinov** [comment]
Any progress with the ability to assign an issue to a team?

**Comment by kilasuit** [comment]
We do this using labels right now but it would be nicer for this to be via teams instead

**Comment by ViacheslavKudinov** [comment]
It can be used (most likely i will use labels and `@team` in a "body") as a workaround, but will be great to get same as we have for PR:s.

**Comment by simon3c** [comment]
+1

**Comment by maxmanus96** [comment]
+1

**Comment by jason-curtis** [comment]
This may be obvious, but I was able to do this only after I allowed the particular team access to the repository.

**Comment by j-refs** [comment]
+4  +7 for both organizations I manage.

7 years, and not a single word from anyone at GitHub? I am genuinely disappointed. Especially since most of us with larger teams actually pay for your lights to stay on...

Like, from a developers viewpoint, the basic functionality is there, in PRs and labeling with @, but simply just not applied to issues. Which makes literally just as much sense as having it on PRs. You usually have an issue before you open a PR. It makes sense for a team to triage the issue, cuz I mean come on, 1 person? If you have assignees automatically set, that's swamping 1 single person with all the issues. Very much against your business philosophy there, GitHub 🤨. It follows sense that the team can pass the issue off to the more appropriate team after triage to actually get fixed, or even handled by the same team, rather than ONE. SINGLE. PERSON. delegating every single issue. I mean, unless you encourage workarounds to get basic functionality? 

I might even be so bold as to say it makes MORE sense for an issue to be assigned to a team rather than a PR. A team to handle the issue, and really only 1 person is needed to manage the PR, either approve/merge or deny/close. 

Come on GitHub. It's clearly an easy to implement feature in high demand with high value.
We know you got the manpower to do it. We know you got the money to do it. You clearly have plenty of time to be sitting on this for 7 years. What is your excuse? Seriously, engineer to engineer, what can possibly excuse NOT EVEN MAKING A COMMENT IN 7 YEARS?

**Comment by cristiklein** [comment]
+1

**Comment by bchen1029** [comment]
+1

**Comment by nunodsfernandes** [comment]
Another +1 for this..
It's sad that after years, no one picks this up.

Large enterprise (100K+) with multiple products.
Assigning issues to teams is paramount for our work.

**Comment by BillChirico** [comment]
Definitely a great feature request +1

**Comment by FynnBe** [comment]
+1

**Comment by rwaffen** [comment]
+1

**Comment by k-doering-NOAA** [comment]
+1

**Comment by paalaas** [comment]
+1

**Comment by paula-jayne** [comment]
+1 this would be great!

**Comment by wietzesuijker** [comment]
+1

**Comment by escolhatecnologia** [comment]
+1

**Comment by j-refs** [comment]
Ummmm, "dear-github', from one dev to another, PM to PM, 7 years with no response on your own Q/A, for a largely requested feature, is (in my honest opinion of course) overly unacceptable. 

Bitbucket already provides this feature, and supports it well. Having been a paying customer of yours for quite a while, I am severely disappointed. Across all the times and ways I've reached out to you, as a paying customer depending on you for my income, I've never gotten even remotely adequate support. I think, that its time I take my business elsewhere. 
I will be moving to bitbucket, and recommend everyone else on this thread seeking basic functionality from a tech giant to do the same. And maybe if your staff get back on the ball, I'll consider being your customer again. But as I stand now, as the companies I represent stand now, we just dont feel respected nor appreciated, and will be taking all my git related business to bitbucket.

And, an honest side note to my other devs out there, this is exactly what happens when Microsoft gets their greedy hands on something. Microsoft is killing github, like its killed many other good things we used to take for granted. I'm awfully glad I dont use Azure, I can't imagine the headaches that would have caused.

**Comment by einaren** [comment]
I finally got curious enough to have a look. It doesn't seem like this repo is connected to GitHub in any way, other than beeing a repo on GitHub? 

Seems like https://github.com/orgs/community/discussions/categories/projects-and-issues is a better place to provide feedback?

**Comment by bobbyphtr** [comment]
Hi.. thanks for the idea @einaren, I forwarded it to the community discussion. Hopefully got lots of support!
https://github.com/orgs/community/discussions/86679

**Comment by bout3fiddy** [comment]
We need this! I think assigning to a team and allowing them team to further handler the sub-assignee makes sense to me.

**Comment by shreya-goddu** [comment]
+1 this would be incredibly helpful to for projects that are worked on across multiple teams

**Comment by dan-myles** [comment]
+1 would be super useful!

**Comment by einaren** [comment]
@dan-myles @shreya-goddu @bout3fiddy Have a look at the comments from Dec 13 and Jan 12 above. There's better chance of reaching the correct people in GitHubs own community site

**Comment by nunodsfernandes** [comment]
Another request for update there.
No one seems to give a single f on GH about community feedback.
Sad...

**Comment by soufianerafik** [comment]
+1 to this feature request! It would be incredibly useful to allow issues to be assigned to a team directly. When team members leave, it becomes challenging to track who should be assigned to issues. Instead of having to reassign individual users, it would be much more efficient to assign issues to a team (a team of developers or a dispatcher) to ensure that the appropriate people are handling the issues.

Hope we get this in the future! Thanks for all the great work you’re doing.

**Comment by hcnp** [comment]
+1

**Comment by benjaminbach-kr** [comment]
+1 crazy that it can't handle this, either the team , or deconstruct the assigned team dynamically and still do individuals listed if they want users to be listed there

**Comment by docktermj** [comment]
In an organization where reviewers change every few years, this would simplify the overhead of changing hundreds of `.github/dependabot.yaml` files.
